src/main/groovy/util/CustomMessageSource.groovy

```diff
@@ -0,0 +1,23 @@
+package util
+
+import org.springframework.context.support.ResourceBundleMessageSource
+import java.text.MessageFormat
+
+class CustomMessageSource extends ResourceBundleMessageSource {
+
+    /**
+     * Custom message source resolver to deal with country specific locales.
+     * For example when we have `es_MX` we want to first fallback to the
+     * language locale -> `es`, and then to system default locale -> `en`
+     * */
+    @Override
+    protected MessageFormat resolveCode(String code, Locale locale) {
```

Comment:
is it a method that is now currently used behind the scenes when resolving the translations?

Reply:
Yes

Reply:
I'm concerned about the side effects, but I love the idea. We can also probably come up with a better name. :)

Reply:
I can't think of a good name myself, but wondering if you have any more insight into what it should be called. Maybe `CascadingResourceBundleMessageSource`?

Otherwise, let's go with CustomResourceBundleMessageSource.

Reply:
Turns out that HierarchicalMessageSource is an interface implemented by ResourceBundleMessageSource. So now my head is spinning :face_with_diagonal_mouth: Does that change anything for you @awalkowiak? Does ResourceBundleMessageSource not do the cascading through the resource bundles?

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/HierarchicalMessageSource.html